tests/system_tests_link_routes.py

```diff
@@ -1797,110 +1814,227 @@ def _fake_broker(self, cls):
+        begin_frame.exit()
+        begin_frame.exit()
+        begin_frame.rewind()
+
+        attach_frame = Data()
+        attach_frame.put_described()
+        attach_frame.enter()
+        attach_frame.put_ulong(self.ATTACH_DESCRIPTOR)
+        attach_frame.put_list()
+        attach_frame.enter()
+        attach_frame.put_string("test-link-name")
+        attach_frame.put_uint(0)      # handle
+        attach_frame.put_bool(False)  # sender
+        attach_frame.put_null()
+        attach_frame.put_null()
+        attach_frame.put_null()
+        # target:
+        attach_frame.put_described()
+        attach_frame.enter()
+        attach_frame.put_ulong(self.TARGET_DESCRIPTOR)
+        attach_frame.put_list()
+        attach_frame.enter()
+        attach_frame.put_string("examples/foo")
+        attach_frame.exit()  # target list
+        attach_frame.exit()  # target descriptor
+        attach_frame.exit()    # attach list
+        attach_frame.exit()    # attach descriptor
+        attach_frame.rewind()
+
+        for frame in [open_frame, begin_frame, attach_frame]:
+            self._send_frame(frame, s)
 
         # Give a second for the attach to propagate to the broker and
-        # for the broker to send a response attach
+        # for the broker to send a response attach and flow:
         sleep(1)
-        data = s.recv(2048)
-        self.assertIn("examples", repr(data))
-
-        # First send a message on link routed address "examples" with
-        # message body of "message 0"
-        # Verify the the sent message has been accepted.
-        TRANSFER_1 = bytearray(b'\x00\x00\x00\x31\x02\x00\x00\x00'
-                               + b'\x00\x53\x14\xc0\x0f\x0b\x43\x52\x01'
-                               + b'\xa0\x01\x01\x43\x42'
-                               + b'\x40\x40\x40\x40\x40\x42\x00\x53'
-                               + b'\x73\xc0\x02\x01\x44\x00\x53\x77'
-                               + b'\xa1\x09\x6d\x65\x73\x73\x61\x67'
-                               + b'\x65\x20\x30')
-        s.sendall(TRANSFER_1)
+        data = s.recv(4096)
+        self.assertEqual(data[:8], b'AMQP\x00\x01\x00\x00')
+        # expect that the connection was accepted: check for a flow frame:
+        flow_frame = self._find_frame(data[8:], self.FLOW_DESCRIPTOR)
+        self.assertIsNotNone(flow_frame)
+
+        # First send a message on link routed address "examples" with a small
+        # message body. Verify the the sent message has been accepted.
+        t1_frame = self._construct_transfer(0, b'\x01', add_ma=True, add_body=True)
+        self._send_frame(t1_frame, s)
+
+        # We expect to get a disposition frame that accepted the message
         sleep(0.5)
         data = s.recv(1024)
-        # The delivery has been accepted.
-        self.assertIn("x00S$E", repr(data))
+        dispo_frame = self._find_frame(data, self.DISPO_DESCRIPTOR)
+        self.assertIsNotNone(dispo_frame, "expected a disposition (none arrived!)")
```

Comment:
This is fragile in case s.recv() receives say only half of the AMQP frame the first time it runs, and the other half the second time, and the expected data is found on the border of both frames. So the suggestion with the retry I tried to give above does not apply, it would make the test flaky here. (Well, the s.recv()s would have to be appending to a buffer, which gets checked in the entirety each time, I guess, so solutions exist.)